### PR TITLE
Aoscene rewrite

### PR DESCRIPTION
**Pull request by skyedeving**
Some various fixes so that the the branch will build now. I don't know if there's proper behaviour though.

### CODE REVIEW

File: src/courtroom.cpp
Comment (oldmud0) [inline src/courtroom.cpp:2273]: This looks to be the result of a bad merge. This should have functionality attached to it - see #356

File: src/courtroom.cpp
Comment (oldmud0) [inline src/courtroom.cpp:2400]: This is cursed

File: src/courtroom.cpp
Comment (oldmud0) [inline src/courtroom.cpp:2427]: I'm... not sure this has to be indented. I suspect it's on the wrong part of the control flow now.

**Review by oldmud0 — CHANGES_REQUESTED** [review]
This is a very cursed pull request

File: src/courtroom.cpp
Comment (in1tiate) [reply src/courtroom.cpp:2400]: agreed however this particular bit of cursedness was there in #302

Comment (in1tiate) [reply src/courtroom.cpp:2400]: i will probably make this a for loop once the merge conflicts are finished conflicting

File: src/courtroom.cpp
Comment (skyedeving) [reply src/courtroom.cpp:2273]: What confuses me is that the functionality does exist--in another function :thinking:  
`log_chatmessage`: https://github.com/AttorneyOnline/AO2-Client/blob/af25625c047fc0cd1ba9a87219503ed532ee1d28/src/courtroom.cpp#L2054

File: src/courtroom.cpp
Comment (skyedeving) [reply src/courtroom.cpp:2427]: For this part it was after adding in missing brackets and trusting my editor to indent. I don't know how the brackets went missing after the merge attempt, but the branch builds now.

File: src/courtroom.cpp
Comment (skyedeving) [reply src/courtroom.cpp:2273]: No wait, I see the issue now, it should be appending here instead of logging

Comment (skyedeving) [reply src/courtroom.cpp:2273]: I take that back, it exists in `display_log_chatmessage`

File: src/courtroom.cpp
Comment (skyedeving) [reply src/courtroom.cpp:2427]: The client seems to work properly, although Ema's gets to eat 2.9 snackoos instead of 1 before talking. Feature or bug? The chat arrow also moved up too.

Comment (in1tiate) [reply src/courtroom.cpp:2427]: i can only assume Ema's snackoo eating was truncated by the [Time] part of the char.ini, which after my rewrite no longer does anything

Comment (in1tiate) [reply src/courtroom.cpp:2427]: the chat_arrow moving up is due to no longer using QMovie to render it, which moves it because Qt is just Like That

**Review by oldmud0 — APPROVED** [review]
Merging this in to unblock the AOScene branch.

**Pull request merged**
